src/utils/collection-stream.js

```diff
@@ -3,97 +3,55 @@ import { Writable } from 'stream';
 import { createLogger } from './logger';
 const debug = createLogger('collection-stream');
 
+function mongodbServerErrorToJSError({ index, code, errmsg, op, errInfo }) {
+  const e = new Error(errmsg);
+  e.index = index;
+  e.code = code;
+  e.op = op;
+  e.errInfo = errInfo;
+  // https://docs.mongodb.com/manual/reference/method/BulkWriteResult/#BulkWriteResult.writeErrors
+  e.name = index && op ? 'WriteError' : 'WriteConcernError';
+  return e;
+}
+
 class WritableCollectionStream extends Writable {
   constructor(dataService, ns, stopOnErrors) {
     super({ objectMode: true });
     this.dataService = dataService;
     this.ns = ns;
     this.BATCH_SIZE = 1000;
     this.docsWritten = 0;
+    this.docsProcessed = 0;
     this.stopOnErrors = stopOnErrors;
 
-    this._initBatch();
+    this.batch = [];
     this._batchCounter = 0;
+
     this._stats = {
+      ok: 0,
       nInserted: 0,
       nMatched: 0,
       nModified: 0,
       nRemoved: 0,
       nUpserted: 0,
-      ok: 0,
-      writeErrorCount: 0
+      writeErrors: [],
+      writeConcernErrors: [],
     };
 
     this._errors = [];
   }
 
-  _initBatch() {
-    this.batch = this._collection().initializeOrderedBulkOp({
-      explicitlyIgnoreSession: true,
-      retryWrites: false,
-      writeConcern: {
-        w: 1
-      },
-      // TODO: lucas: option in mongoimport w slightly different name?
-      checkKeys: false
-    });
-  }
-
   _collection() {
     return this.dataService.client._collection(this.ns);
   }
 
-  _write(chunk, encoding, next) {
-    this.batch.insert(chunk);
-    if (this.batch.length === this.BATCH_SIZE) {
-      // TODO: lucas: expose finer-grained bulk op results:
-      // https://mongodb.github.io/node-mongodb-native/3.3/api/BulkWriteResult.html
-      const nextBatch = (err, res = {}) => {
-        if (err && this.stopOnErrors) {
-          return next(err);
-        }
-        this.docsWritten += this.batch.length;
-        if (err) {
-          debug(`batch ${this._batchCounter} result`, { err, res });
-        }
-        this.captureStatsForBulkResult(err, res);
+  _write(document, _encoding, next) {
+    this.batch.push(document);
 
-        this._batchCounter++;
-        this._initBatch();
-        next();
-      };
-
-      const execBatch = cb => {
-        const batchSize = this.batch.length;
-        this.batch.execute((err, res) => {
-          /**
-           * TODO: lucas: appears turning off retyableWrites
-           * gives a slightly different error but probably same problem?
-           */
-          if (
-            err &&
-            Array.isArray(err.errorLabels) &&
-            err.errorLabels.indexOf('TransientTransactionError')
-          ) {
-            err = null;
-            res = { nInserted: batchSize };
-          }
```

Comment:
Found no signs of this error being possible when using `bulkWrite`. I would rather remove this for now and bring it back only if someone reports this (my bet is that this will not happen)